src/puppetlabs/puppetdb/http/command.clj

```diff
@@ -77,40 +92,145 @@
           timeout-chan (http/json-response {:uuid uuid
                                             :processed false
                                             :timed_out true}
-                                           503)
+                                           http/status-unavailable)
           response-chan ([{:keys [command exception]}]
                          (let [base-response {:uuid uuid
                                               :processed true}]
                            (if exception
-                             (http/json-response (assoc base-response
-                                                        :timed_out false
-                                                        :error (str exception)
-                                                        :stack_trace (map str (.getStackTrace exception)))
-                                                 503)
+                             (http/json-response
+                              (assoc base-response
+                                     :timed_out false
+                                     :error (str exception)
+                                     :stack_trace (map str (.getStackTrace exception)))
+                              http/status-unavailable)
                              (http/json-response (assoc base-response
                                                         :timed_out false)
-                                                 200)))))))))
+                                                 http/status-ok)))))))))
+
+(defn- normalize-request
+  "Converts request to the \"one true format\" if possible.  Ensures
+  that the request :params include \"command\", \"version\", and maybe
+  \"certname\" entries, and that the entries are in the correct
+  format, i.e. spaces instead of underscores in the command name,
+  integer instead of string for the version, etc.  Ensures that
+  the :body only contains the \"payload\", as either a string or
+  stream.  The :body will be a stream unless reading the body stream
+  is unavoidable (i.e. old-style, non-param POST)."
+  [handle]
+  (fn [{:keys [body params] :as req}]
+    (let [req (if-let [cmd (params "command")]
+                (assoc-in req [:params "command"] (str/replace cmd "_" " "))
+                req)
+          req (if-let [ver (params "version")]
+                (assoc-in req [:params "version"] (Integer/parseInt ver))
+                req)
+          {:strs [command version certname]} params
```

Comment:
Hmm, what did you have in mind -- the intention was for them to have the "final" values, after any fixup.